metrics/api/src/main/java/io/helidon/metrics/api/spi/HelidonMetricFactory.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.metrics.api.spi;
+
+import java.util.function.Function;
+import java.util.function.ToDoubleFunction;
+
+import io.helidon.metrics.api.Clock;
+import io.helidon.metrics.api.Counter;
+import io.helidon.metrics.api.DistributionSummary;
+import io.helidon.metrics.api.HistogramSnapshot;
+import io.helidon.metrics.api.MeterRegistry;
+import io.helidon.metrics.api.Tag;
+import io.helidon.metrics.api.Timer;
+
+/**
+ * Factory for creating implementation instances of the Helidon metrics API.
+ * <p>
+ *     An implementation of this interface provides instance methods for each
+ *     of the static methods on the Helidon metrics API interfaces. The prefix of each method
+ *     here identifies the interface that bears the corresponding static method. For example,
+ *     {@link #timerStart(io.helidon.metrics.api.MeterRegistry)} corresponds to the static
+ *     {@link Timer#start(io.helidon.metrics.api.MeterRegistry)} method.
+ * </p>
+ */
+public interface HelidonMetricFactory {
```

Comment:
Why is this SPI? Is it to be implemented as a `ServiceProvider`? If so, it should be named `MetricFactoryProvider`.
Also please use package `io.helidon.metrics.spi` - we have agreed on this exception to the rule

Reply:
I actually need to revise this a bit anyway so the implementation can accept configuration (or a `Supplier` of configuration) which might influence how metrics should behave. I'll incorporate your comments.